diff --git a/include/mode-s.h b/include/mode-s.h
index 1c4d326..f8dd2bf 100644
--- a/include/mode-s.h
+++ b/include/mode-s.h
@@ -4,9 +4,16 @@
 #include <string.h>
 #include <stdlib.h>
 #include <stdint.h>
-#include <unistd.h>
 #include <math.h>
-#include <sys/time.h>
+#include <time.h>
+
+#ifndef _WIN32
+#include <unistd.h>
+#include <sys/time.h>
+#endif
+
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
 
 #define MODE_S_ICAO_CACHE_LEN 1024 // Power of two required
 #define MODE_S_LONG_MSG_BYTES (112/8)
